include/tvm/runtime/vm/bytecode.h

```diff
@@ -204,6 +207,13 @@ struct Instruction {
       RegName tensor;
       RegName newshape;
     } reshape_tensor;
+    struct /* DeviceCopy Operands */ {
+      RegName src;
+      /*! \brief The source device type. */
+      Index src_device_type;
```

Comment:
should it be device type or device id?

Reply:
It should be device type (i.e. the source is kDLCPU and destination is kDLGPU). We haven't really effectively use device id so far.

Reply:
But it limits the use of two devices with the same type, e.g., two GPUs.

Reply:
As per offline discussion, we can keep device type for now since this is not a quite typical case for inference. We can have a separate PR to use device id in VM if it is needed in the future.